### ISSUE TITLE
Fulltext search causes solr error

### ISSUE THREAD

**Issue by J0n4t4n** [eikek/docspell#609]
Hey, I discovered your project recently and so far it seems to be the document management software that I have been looking for for a long time.

I have deployed the docker setup and the only things I changed are:
1. Change the mount points to folders
2. Move container in their own docker network
3. Nginx Reverse proxy as described in docs

I have 4 PDFs uploaded currently.
When I try to use the Fulltext / Content search nothing happens and this error is thrown by the solr container:

```
docspell-restserver | [ioapp-compute-6] INFO  d.r.r.ItemRoutes - Make index only summary: ItemFtsSearch(0,60,DHBW) 
docspell-solr | 2021-01-29 14:31:16.406 ERROR (qtp691691381-23) [   x:docspell] o.a.s.h.RequestHandlerBase org.apache.solr.common.SolrException: undefined field content
docspell-solr | 	at org.apache.solr.schema.IndexSchema.getDynamicFieldType(IndexSchema.java:1325)
docspell-solr | 	at org.apache.solr.schema.IndexSchema$SolrQueryAnalyzer.getWrappedAnalyzer(IndexSchema.java:473)
docspell-solr | 	at org.apache.lucene.analysis.DelegatingAnalyzerWrapper$DelegatingReuseStrategy.getReusableComponents(DelegatingAnalyzerWrapper.java:86)
docspell-solr | 	at org.apache.lucene.analysis.Analyzer.tokenStream(Analyzer.java:192)
docspell-solr | 	at org.apache.lucene.util.QueryBuilder.createFieldQuery(QueryBuilder.java:262)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.newFieldQuery(SolrQueryParserBase.java:534)
docspell-solr | 	at org.apache.solr.parser.QueryParser.newFieldQuery(QueryParser.java:62)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.getFieldQuery(SolrQueryParserBase.java:1109)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.handleBareTokenQuery(SolrQueryParserBase.java:832)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Term(QueryParser.java:421)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:278)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:282)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
docspell-solr | 	at org.apache.solr.parser.QueryParser.TopLevelQuery(QueryParser.java:131)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.parse(SolrQueryParserBase.java:270)
docspell-solr | 	at org.apache.solr.search.LuceneQParser.parse(LuceneQParser.java:49)
docspell-solr | 	at org.apache.solr.search.QParser.getQuery(QParser.java:174)
docspell-solr | 	at org.apache.solr.handler.component.QueryComponent.prepare(QueryComponent.java:162)
docspell-solr | 	at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:337)
docspell-solr | 	at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:214)
docspell-solr | 	at org.apache.solr.core.SolrCore.execute(SolrCore.java:2627)
docspell-solr | 	at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:795)
docspell-solr | 	at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:568)
docspell-solr | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:415)
docspell-solr | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:345)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1596)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:545)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
docspell-solr | 	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:590)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
docspell-solr | 	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1300)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:485)
docspell-solr | 	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1215)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:221)
docspell-solr | 	at org.eclipse.jetty.server.handler.InetAccessHandler.handle(InetAccessHandler.java:177)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:322)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.server.Server.handle(Server.java:500)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
docspell-solr | 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
docspell-solr | 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
docspell-solr | 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
docspell-solr | 	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
docspell-solr | 	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
docspell-solr | 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
docspell-solr | 	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
docspell-solr | 	at java.base/java.lang.Thread.run(Unknown Source)
docspell-solr | 
docspell-solr | 2021-01-29 14:31:16.406 INFO  (qtp691691381-23) [   x:docspell] o.a.s.c.S.Request [docspell]  webapp=/solr path=/query params={} status=400 QTime=3
docspell-solr | 2021-01-29 14:31:16.410 ERROR (qtp691691381-25) [   x:docspell] o.a.s.h.RequestHandlerBase org.apache.solr.common.SolrException: undefined field content
docspell-solr | 	at org.apache.solr.schema.IndexSchema.getDynamicFieldType(IndexSchema.java:1325)
docspell-solr | 	at org.apache.solr.schema.IndexSchema$SolrQueryAnalyzer.getWrappedAnalyzer(IndexSchema.java:473)
docspell-solr | 	at org.apache.lucene.analysis.DelegatingAnalyzerWrapper$DelegatingReuseStrategy.getReusableComponents(DelegatingAnalyzerWrapper.java:86)
docspell-solr | 	at org.apache.lucene.analysis.Analyzer.tokenStream(Analyzer.java:192)
docspell-solr | 	at org.apache.lucene.util.QueryBuilder.createFieldQuery(QueryBuilder.java:262)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.newFieldQuery(SolrQueryParserBase.java:534)
docspell-solr | 	at org.apache.solr.parser.QueryParser.newFieldQuery(QueryParser.java:62)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.getFieldQuery(SolrQueryParserBase.java:1109)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.handleBareTokenQuery(SolrQueryParserBase.java:832)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Term(QueryParser.java:421)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:278)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:282)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
docspell-solr | 	at org.apache.solr.parser.QueryParser.TopLevelQuery(QueryParser.java:131)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.parse(SolrQueryParserBase.java:270)
docspell-solr | 	at org.apache.solr.search.LuceneQParser.parse(LuceneQParser.java:49)
docspell-solr | 	at org.apache.solr.search.QParser.getQuery(QParser.java:174)
docspell-solr | 	at org.apache.solr.handler.component.QueryComponent.prepare(QueryComponent.java:162)
docspell-solr | 	at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:337)
docspell-solr | 	at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:214)
docspell-solr | 	at org.apache.solr.core.SolrCore.execute(SolrCore.java:2627)
docspell-solr | 	at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:795)
docspell-solr | 	at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:568)
docspell-solr | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:415)
docspell-solr | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:345)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1596)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:545)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
docspell-solr | 	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:590)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
docspell-solr | 	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1300)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:485)
docspell-solr | 	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1215)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:221)
docspell-solr | 	at org.eclipse.jetty.server.handler.InetAccessHandler.handle(InetAccessHandler.java:177)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:322)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.server.Server.handle(Server.java:500)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
docspell-solr | 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
docspell-solr | 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
docspell-solr | 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
docspell-solr | 	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
docspell-solr | 	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
docspell-solr | 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
docspell-solr | 	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
docspell-solr | 	at java.base/java.lang.Thread.run(Unknown Source)
docspell-solr | 
docspell-solr | 2021-01-29 14:31:16.410 INFO  (qtp691691381-25) [   x:docspell] o.a.s.c.S.Request [docspell]  webapp=/solr path=/query params={} status=400 QTime=8
docspell-restserver | [docspell-restserver-blocking-36] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/sec/item/searchWithTags 
docspell-restserver | [docspell-restserver-blocking-36] INFO  o.h.s.m.Logger - service raised an error: class org.http4s.client.UnexpectedStatus 
docspell-restserver | [docspell-restserver-blocking-36] ERROR o.h.s.service-errors - Error servicing request: POST /api/v1/sec/item/searchWithTags from 172.28.0.1 
docspell-restserver | org.http4s.client.UnexpectedStatus: unexpected HTTP status: 400 Bad Request
docspell-restserver | [docspell-restserver-blocking-33] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/sec/item/searchStats 
docspell-restserver | [docspell-restserver-blocking-33] INFO  o.h.s.m.Logger - service raised an error: class org.http4s.client.UnexpectedStatus 
docspell-restserver | [docspell-restserver-blocking-33] ERROR o.h.s.service-errors - Error servicing request: POST /api/v1/sec/item/searchStats from 172.28.0.1 
docspell-restserver | org.http4s.client.UnexpectedStatus: unexpected HTTP status: 400 Bad Request
```

**Comment by eikek** [comment]
Hi @J0n4t4n  and thanks!

This looks like solr is not setup. Docspell does this one startup, but only once – subsequent startups will not initiate this setup again. There is already an issue to improve on that (#604). In the meantime, you can try to use the admin endpoint to initiate a re-index which also runs the setup again. This is described [here](https://docspell.org/docs/configure/#full-text-search-solr). You need to set an admin password in the config file to enable this endpoint. Another way is to empty the table `fts_migration` and then restart docspell.

**Comment by J0n4t4n** [comment]
Ah yes, that makes sense. I had some trouble with my folder mount and the permissions in the beginning so the initial setup probably got lost.

I‘ll try a re-index as you described and report back.

**Comment by J0n4t4n** [comment]
So I enabled the admin endpoint and was able to trigger the re-index with the curl command.
Sadly it didn't work and I still get the same error.

This is the log from the reindex and my search try.
It seems to want to connect to Joex with some seemingly odd hostnames, if that might have something to do with it.

```
docspell-restserver | [blaze-selector-7] INFO  o.h.b.c.n.NIO1SocketServerGroup - Accepted connection from /172.28.0.1:41346 
docspell-restserver | [ioapp-compute-2] INFO  d.b.o.OFulltext - Re-index all. 
docspell-restserver | [ioapp-compute-4] DEBUG d.j.c.JoexClient - Notify joex at http://12a4774cbfc1:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-3] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://12a4774cbfc1:7878' failed: Error connecting to http://12a4774cbfc1:7878 using address 12a4774cbfc1:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-3] DEBUG d.j.c.JoexClient - Notify joex at http://a424f04c6b38:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-2] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://a424f04c6b38:7878' failed: Error connecting to http://a424f04c6b38:7878 using address a424f04c6b38:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-2] DEBUG d.j.c.JoexClient - Notify joex at http://e828c8194dc2:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-2] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://e828c8194dc2:7878' failed: Error connecting to http://e828c8194dc2:7878 using address e828c8194dc2:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-2] DEBUG d.j.c.JoexClient - Notify joex at http://c7f210e1893e:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-2] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://c7f210e1893e:7878' failed: Error connecting to http://c7f210e1893e:7878 using address c7f210e1893e:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-2] DEBUG d.j.c.JoexClient - Notify joex at http://9a60fa5b4b41:7878/api/v1/notify 
joex_1        | [ioapp-compute-4] DEBUG d.j.s.SchedulerImpl - Notify signal, going into main loop 
joex_1        | [ioapp-compute-7] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/notify 
joex_1        | [ioapp-compute-7] INFO  o.h.s.m.Logger - HTTP/1.1 200 OK 
joex_1        | [ioapp-compute-4] DEBUG d.j.s.SchedulerImpl - Try to acquire permit (1 free) 
joex_1        | [ioapp-compute-4] DEBUG d.j.s.SchedulerImpl - New permit acquired 
joex_1        | [ioapp-compute-7] DEBUG d.j.s.PeriodicSchedulerImpl - Notify signal, going into main loop 
joex_1        | [ioapp-compute-7] DEBUG d.j.s.PeriodicSchedulerImpl - Going into main loop 
joex_1        | [ioapp-compute-7] DEBUG d.j.s.PeriodicSchedulerImpl - Looking for next periodic task 
docspell-restserver | [docspell-restserver-blocking-2] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/admin/fts/reIndexAll 
docspell-restserver | [docspell-restserver-blocking-2] INFO  o.h.s.m.Logger - HTTP/1.1 200 OK 
joex_1        | [ioapp-compute-6] DEBUG d.j.s.SchedulerImpl - Next job found: None 
joex_1        | [ioapp-compute-6] DEBUG d.j.s.SchedulerImpl - Waiting for notify 
joex_1        | [ioapp-compute-2] DEBUG d.j.s.PeriodicSchedulerImpl - Found periodic task 'Docspell house-keeping/Sun *-*-* 00:00:00' 
joex_1        | [ioapp-compute-2] DEBUG d.j.s.PeriodicSchedulerImpl - Scheduling next notify for timer Sun *-*-* 00:00:00 -> Some(2021-01-31T00:00) 
joex_1        | [ioapp-compute-7] DEBUG d.j.s.PeriodicSchedulerImpl - Waiting for notify 
docspell-solr | 2021-01-30 14:38:08.526 INFO  (qtp649769713-19) [   x:docspell] o.a.s.c.S.Request [docspell]  webapp=/solr path=/admin/ping params={} hits=0 status=0 QTime=24
docspell-solr | 2021-01-30 14:38:08.526 INFO  (qtp649769713-19) [   x:docspell] o.a.s.c.S.Request [docspell]  webapp=/solr path=/admin/ping params={} status=0 QTime=24
docspell-restserver | [blaze-selector-8] INFO  o.h.b.c.n.NIO1SocketServerGroup - Accepted connection from /127.0.0.1:49100 
docspell-restserver | [ioapp-compute-0] INFO  o.h.s.m.Logger - HTTP/1.1 GET / 
docspell-restserver | [ioapp-compute-0] INFO  o.h.s.m.Logger - HTTP/1.1 303 See Other 
docspell-restserver | [blaze-selector-0] INFO  o.h.b.c.n.NIO1SocketServerGroup - Accepted connection from /127.0.0.1:49102 
docspell-restserver | [ioapp-compute-5] INFO  o.h.s.m.Logger - HTTP/1.1 GET /app 
docspell-restserver | [ioapp-compute-5] INFO  o.h.s.m.Logger - HTTP/1.1 200 OK 
docspell-restserver | [blaze-selector-2] INFO  o.h.b.c.n.NIO1SocketServerGroup - Accepted connection from /172.28.0.1:41380 
docspell-restserver | [blaze-selector-1] INFO  o.h.b.c.n.NIO1SocketServerGroup - Accepted connection from /172.28.0.1:41382 
docspell-restserver | [ioapp-compute-1] INFO  d.r.r.ItemRoutes - Make index only summary: ItemFtsSearch(0,60,DFHBW) 
docspell-solr | 2021-01-30 14:38:17.834 ERROR (qtp649769713-25) [   x:docspell] o.a.s.h.RequestHandlerBase org.apache.solr.common.SolrException: undefined field content
docspell-solr | 	at org.apache.solr.schema.IndexSchema.getDynamicFieldType(IndexSchema.java:1325)
docspell-solr | 	at org.apache.solr.schema.IndexSchema$SolrQueryAnalyzer.getWrappedAnalyzer(IndexSchema.java:473)
docspell-solr | 	at org.apache.lucene.analysis.DelegatingAnalyzerWrapper$DelegatingReuseStrategy.getReusableComponents(DelegatingAnalyzerWrapper.java:86)
docspell-solr | 	at org.apache.lucene.analysis.Analyzer.tokenStream(Analyzer.java:192)
docspell-solr | 	at org.apache.lucene.util.QueryBuilder.createFieldQuery(QueryBuilder.java:262)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.newFieldQuery(SolrQueryParserBase.java:534)
docspell-solr | 	at org.apache.solr.parser.QueryParser.newFieldQuery(QueryParser.java:62)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.getFieldQuery(SolrQueryParserBase.java:1109)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.handleBareTokenQuery(SolrQueryParserBase.java:832)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Term(QueryParser.java:421)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:278)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:282)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
docspell-solr | 	at org.apache.solr.parser.QueryParser.TopLevelQuery(QueryParser.java:131)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.parse(SolrQueryParserBase.java:270)
docspell-solr | 	at org.apache.solr.search.LuceneQParser.parse(LuceneQParser.java:49)
docspell-solr | 	at org.apache.solr.search.QParser.getQuery(QParser.java:174)
docspell-solr | 	at org.apache.solr.handler.component.QueryComponent.prepare(QueryComponent.java:162)
docspell-solr | 	at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:337)
docspell-solr | 	at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:214)
docspell-solr | 	at org.apache.solr.core.SolrCore.execute(SolrCore.java:2627)
docspell-solr | 	at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:795)
docspell-solr | 	at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:568)
docspell-solr | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:415)
docspell-solr | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:345)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1596)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:545)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
docspell-solr | 	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:590)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
docspell-solr | 	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1300)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:485)
docspell-solr | 	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1215)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:221)
docspell-solr | 	at org.eclipse.jetty.server.handler.InetAccessHandler.handle(InetAccessHandler.java:177)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:322)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.server.Server.handle(Server.java:500)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
docspell-solr | 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
docspell-solr | 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
docspell-solr | 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
docspell-solr | 	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
docspell-solr | 	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
docspell-solr | 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
docspell-solr | 	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
docspell-solr | 	at java.base/java.lang.Thread.run(Unknown Source)
docspell-solr | 
docspell-solr | 2021-01-30 14:38:17.835 INFO  (qtp649769713-25) [   x:docspell] o.a.s.c.S.Request [docspell]  webapp=/solr path=/query params={} status=400 QTime=2
docspell-restserver | [docspell-restserver-blocking-2] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/sec/item/searchWithTags 
docspell-restserver | [docspell-restserver-blocking-2] INFO  o.h.s.m.Logger - service raised an error: class org.http4s.client.UnexpectedStatus 
docspell-restserver | [docspell-restserver-blocking-2] ERROR o.h.s.service-errors - Error servicing request: POST /api/v1/sec/item/searchWithTags from 172.28.0.1 
docspell-restserver | org.http4s.client.UnexpectedStatus: unexpected HTTP status: 400 Bad Request
docspell-solr | 2021-01-30 14:38:17.866 ERROR (qtp649769713-22) [   x:docspell] o.a.s.h.RequestHandlerBase org.apache.solr.common.SolrException: undefined field content
docspell-solr | 	at org.apache.solr.schema.IndexSchema.getDynamicFieldType(IndexSchema.java:1325)
docspell-solr | 	at org.apache.solr.schema.IndexSchema$SolrQueryAnalyzer.getWrappedAnalyzer(IndexSchema.java:473)
docspell-solr | 	at org.apache.lucene.analysis.DelegatingAnalyzerWrapper$DelegatingReuseStrategy.getReusableComponents(DelegatingAnalyzerWrapper.java:86)
docspell-solr | 	at org.apache.lucene.analysis.Analyzer.tokenStream(Analyzer.java:192)
docspell-solr | 	at org.apache.lucene.util.QueryBuilder.createFieldQuery(QueryBuilder.java:262)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.newFieldQuery(SolrQueryParserBase.java:534)
docspell-solr | 	at org.apache.solr.parser.QueryParser.newFieldQuery(QueryParser.java:62)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.getFieldQuery(SolrQueryParserBase.java:1109)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.handleBareTokenQuery(SolrQueryParserBase.java:832)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Term(QueryParser.java:421)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:278)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Clause(QueryParser.java:282)
docspell-solr | 	at org.apache.solr.parser.QueryParser.Query(QueryParser.java:162)
docspell-solr | 	at org.apache.solr.parser.QueryParser.TopLevelQuery(QueryParser.java:131)
docspell-solr | 	at org.apache.solr.parser.SolrQueryParserBase.parse(SolrQueryParserBase.java:270)
docspell-solr | 	at org.apache.solr.search.LuceneQParser.parse(LuceneQParser.java:49)
docspell-solr | 	at org.apache.solr.search.QParser.getQuery(QParser.java:174)
docspell-solr | 	at org.apache.solr.handler.component.QueryComponent.prepare(QueryComponent.java:162)
docspell-solr | 	at org.apache.solr.handler.component.SearchHandler.handleRequestBody(SearchHandler.java:337)
docspell-solr | 	at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:214)
docspell-solr | 	at org.apache.solr.core.SolrCore.execute(SolrCore.java:2627)
docspell-solr | 	at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:795)
docspell-solr | 	at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:568)
docspell-solr | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:415)
docspell-solr | 	at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:345)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1596)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:545)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:143)
docspell-solr | 	at org.eclipse.jetty.security.SecurityHandler.handle(SecurityHandler.java:590)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:235)
docspell-solr | 	at org.eclipse.jetty.server.session.SessionHandler.doHandle(SessionHandler.java:1610)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextHandle(ScopedHandler.java:233)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1300)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:188)
docspell-solr | 	at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:485)
docspell-solr | 	at org.eclipse.jetty.server.session.SessionHandler.doScope(SessionHandler.java:1580)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.nextScope(ScopedHandler.java:186)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1215)
docspell-solr | 	at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
docspell-solr | 	at org.eclipse.jetty.server.handler.ContextHandlerCollection.handle(ContextHandlerCollection.java:221)
docspell-solr | 	at org.eclipse.jetty.server.handler.InetAccessHandler.handle(InetAccessHandler.java:177)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerCollection.handle(HandlerCollection.java:146)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.rewrite.handler.RewriteHandler.handle(RewriteHandler.java:322)
docspell-solr | 	at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:127)
docspell-solr | 	at org.eclipse.jetty.server.Server.handle(Server.java:500)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.lambda$handle$1(HttpChannel.java:383)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.dispatch(HttpChannel.java:547)
docspell-solr | 	at org.eclipse.jetty.server.HttpChannel.handle(HttpChannel.java:375)
docspell-solr | 	at org.eclipse.jetty.server.HttpConnection.onFillable(HttpConnection.java:273)
docspell-solr | 	at org.eclipse.jetty.io.AbstractConnection$ReadCallback.succeeded(AbstractConnection.java:311)
docspell-solr | 	at org.eclipse.jetty.io.FillInterest.fillable(FillInterest.java:103)
docspell-solr | 	at org.eclipse.jetty.io.ChannelEndPoint$2.run(ChannelEndPoint.java:117)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.runTask(EatWhatYouKill.java:336)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.doProduce(EatWhatYouKill.java:313)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.tryProduce(EatWhatYouKill.java:171)
docspell-solr | 	at org.eclipse.jetty.util.thread.strategy.EatWhatYouKill.run(EatWhatYouKill.java:129)
docspell-solr | 	at org.eclipse.jetty.util.thread.ReservedThreadExecutor$ReservedThread.run(ReservedThreadExecutor.java:375)
docspell-solr | 	at org.eclipse.jetty.util.thread.QueuedThreadPool.runJob(QueuedThreadPool.java:806)
docspell-solr | 	at org.eclipse.jetty.util.thread.QueuedThreadPool$Runner.run(QueuedThreadPool.java:938)
docspell-solr | 	at java.base/java.lang.Thread.run(Unknown Source)
docspell-solr | 
docspell-solr | 2021-01-30 14:38:17.867 INFO  (qtp649769713-22) [   x:docspell] o.a.s.c.S.Request [docspell]  webapp=/solr path=/query params={} status=400 QTime=1
docspell-restserver | [docspell-restserver-blocking-3] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/sec/item/searchStats 
docspell-restserver | [docspell-restserver-blocking-3] INFO  o.h.s.m.Logger - service raised an error: class org.http4s.client.UnexpectedStatus 
docspell-restserver | [docspell-restserver-blocking-3] ERROR o.h.s.service-errors - Error servicing request: POST /api/v1/sec/item/searchStats from 172.28.0.1 
docspell-restserver | org.http4s.client.UnexpectedStatus: unexpected HTTP status: 400 Bad Request
```

**Comment by eikek** [comment]
Hm strange. From the logs it seems the re-index task isn't executed at all. It is also strange that `fts_migration` is empty. Then docspell should try to setup solr on start. This didn't seem to happen. What version of sorl are you using?

The weird hostname come from the default config used in docker-compose, it may be a problem since joex doesn't know that there is some task (the re-index) to execute. But I suspect that these are nodes not available anymore. So I'd say we fix that first, so that the reindex task really runs. For this, you need to look into the `docspell.conf` file and set `base-url` in `docspell.joex` to some url that is reachable from the restserver container. By default it should be resolvable, but I don't know your setup. Have a look at #608 – the problem is reported there.

Can you then execute the re-index again and capture the logs from joex? Do you see any errors on startup (joex)?

**Comment by J0n4t4n** [comment]
Solr version is set to 8, thus it is using the latest 8.7.0.

Set the hostname to a static value that works and connects successfully at the last attempt.
Looking at the last try it also seems like it fails multiple times but as the last one doesn't have the connection error, it seems like it was able to connect in the end.

No startup errors from joex or any other container that I can identify.

Re-index done, search still broken, log right here:

```
docspell-restserver | [blaze-selector-2] INFO  o.h.b.c.n.NIO1SocketServerGroup - Accepted connection from /172.29.0.1:33878 
docspell-restserver | [ioapp-compute-6] INFO  d.b.o.OFulltext - Re-index all. 
docspell-restserver | [ioapp-compute-7] DEBUG d.j.c.JoexClient - Notify joex at http://12a4774cbfc1:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-0] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://12a4774cbfc1:7878' failed: Error connecting to http://12a4774cbfc1:7878 using address 12a4774cbfc1:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-0] DEBUG d.j.c.JoexClient - Notify joex at http://a424f04c6b38:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-0] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://a424f04c6b38:7878' failed: Error connecting to http://a424f04c6b38:7878 using address a424f04c6b38:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-0] DEBUG d.j.c.JoexClient - Notify joex at http://e828c8194dc2:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-0] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://e828c8194dc2:7878' failed: Error connecting to http://e828c8194dc2:7878 using address e828c8194dc2:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-0] DEBUG d.j.c.JoexClient - Notify joex at http://c7f210e1893e:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-0] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://c7f210e1893e:7878' failed: Error connecting to http://c7f210e1893e:7878 using address c7f210e1893e:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-0] DEBUG d.j.c.JoexClient - Notify joex at http://9a60fa5b4b41:7878/api/v1/notify 
docspell-restserver | [docspell-restserver-blocking-0] WARN  d.j.c.JoexClient - Notifying Joex instance 'http://9a60fa5b4b41:7878' failed: Error connecting to http://9a60fa5b4b41:7878 using address 9a60fa5b4b41:7878 (unresolved: true) 
docspell-restserver | [docspell-restserver-blocking-0] DEBUG d.j.c.JoexClient - Notify joex at http://docker_joex_1:7878/api/v1/notify 
joex_1        | [blaze-selector-3] INFO  o.h.b.c.n.NIO1SocketServerGroup - Accepted connection from /172.29.0.4:51314 
joex_1        | [ioapp-compute-0] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/notify 
joex_1        | [ioapp-compute-0] INFO  o.h.s.m.Logger - HTTP/1.1 200 OK 
joex_1        | [ioapp-compute-2] DEBUG d.j.s.SchedulerImpl - Notify signal, going into main loop 
joex_1        | [ioapp-compute-2] DEBUG d.j.s.SchedulerImpl - Try to acquire permit (1 free) 
joex_1        | [ioapp-compute-7] DEBUG d.j.s.PeriodicSchedulerImpl - Notify signal, going into main loop 
joex_1        | [ioapp-compute-2] DEBUG d.j.s.SchedulerImpl - New permit acquired 
joex_1        | [ioapp-compute-7] DEBUG d.j.s.PeriodicSchedulerImpl - Going into main loop 
joex_1        | [ioapp-compute-7] DEBUG d.j.s.PeriodicSchedulerImpl - Looking for next periodic task 
docspell-restserver | [docspell-restserver-blocking-0] INFO  o.h.s.m.Logger - HTTP/1.1 POST /api/v1/admin/fts/reIndexAll 
docspell-restserver | [docspell-restserver-blocking-0] INFO  o.h.s.m.Logger - HTTP/1.1 200 OK 
joex_1        | [ioapp-compute-1] DEBUG d.j.s.SchedulerImpl - Next job found: None 
joex_1        | [ioapp-compute-1] DEBUG d.j.s.SchedulerImpl - Waiting for notify 
joex_1        | [ioapp-compute-6] DEBUG d.j.s.PeriodicSchedulerImpl - Found periodic task 'Docspell house-keeping/Sun *-*-* 00:00:00' 
joex_1        | [ioapp-compute-6] DEBUG d.j.s.PeriodicSchedulerImpl - Scheduling next notify for timer Sun *-*-* 00:00:00 -> Some(2021-01-31T00:00) 
joex_1        | [ioapp-compute-6] DEBUG d.j.s.PeriodicSchedulerImpl - Waiting for notify
```

**Comment by eikek** [comment]
Ok, still can't explain what's going on :-/. When the reindex is running, you should see lots of solr log output. But there is none. Also it is really strange that `fts_migration` is empty and docspell doesn't run the schema migration on start.

When I start reindex here, I have logs like these:
```
joex [ioapp-compute-3] INFO  d.j.s.LogSink - >>> 2021-01-30T16:22:38.386Z Info F32z4ZiZe.../docspell-system/full-text-reindex/Low: Running full-text re-index now
joex [ioapp-compute-2] INFO  d.j.s.LogSink - >>> 2021-01-30T16:22:38.386Z Info F32z4ZiZe.../docspell-system/full-text-reindex/Low: Clearing index data
joex [ioapp-compute-2] INFO  d.j.s.LogSink - >>> 2021-01-30T16:22:38.386Z Info F32z4ZiZe.../docspell-system/full-text-reindex/Low: Deleting complete full-text index!
joex [ioapp-compute-6] INFO  d.j.s.LogSink - >>> 2021-01-30T16:22:38.406Z Info F32z4ZiZe.../docspell-system/full-text-reindex/Low: Running index initialize
joex [ioapp-compute-4] DEBUG d.j.s.LogSink - >>> 2021-01-30T16:22:46.069Z Debug F32z4ZiZe.../docspell-system/full-text-reindex/Low: Updating SOLR index
joex [ioapp-compute-0] DEBUG d.j.s.LogSink - >>> 2021-01-30T16:22:46.088Z Debug F32z4ZiZe.../docspell-system/full-text-reindex/Low: Updating SOLR index
joex [ioapp-compute-6] INFO  d.j.s.SchedulerImpl - Job execution successful: F32z4ZiZe.../docspell-system/full-text-reindex/Low
joex [ioapp-compute-6] DEBUG d.j.s.SchedulerImpl - Job F32z4ZiZe.../docspell-system/full-text-reindex/Low done Success. Releasing resources.
```
Do you see any thing similar in your case?

Could you run this sql?
```
select state,args,task,started,finished from job where task = 'full-text-reindex';
```

To get to the job specific logs better, there is a "trick":) Register a new collective 'docspell-system' and log in with that. Then you see the system jobs in the processing page and can view logs specific to this reindex task for example. Hope to find out more….

**Comment by J0n4t4n** [comment]
I sadly don't see any output like that in the log 😅

SQL result also empty:
```
root@325f9f19ce04:/# psql -U dbuser -h localhost -d dbname
psql (11.7 (Debian 11.7-2.pgdg90+1))
Type "help" for help.

dbname=# select state,args,task,started,finished from job where task = 'full-text-reindex';
 state | args | task | started | finished 
-------+------+------+---------+----------
(0 rows)
```

The docspell-system processing page only displays "Create all page-counts" and "Create preview images" events, even after triggering a re-index.

I also tried reverting some of my custom changes, like putting the containers in their own docker network.
I tried changing the passwords I use for admin and header_value to passwords without special characters and in the process found out that I had 2 header_values in my env, fixed that but still no luck :/

I'm officially also confused :D

**Comment by eikek** [comment]
Oh, well. It is really weird. But at least it shows that the reindex task doesn't even arrive in the job queue. When there are no errors in the logs, I can only guess that you might be looking at a different db? Do all your nodes connect to the same db – could you verify this? And are the solr config the same for joex and restserver? I have otherwise no idea currently that could explain tthis….

**Comment by eikek** [comment]
Hi @J0n4t4n , just want to get back to ask whether you could solve this? Or need more support – I'm glad to help if I can.

**Comment by TheOneValen** [comment]
I have the same Problem. I tried reindexing from web-ui (got confirmed) to no avail :(

**Comment by eikek** [comment]
hi @TheOneValen , did you try the admin endpoint for reindexing? if not, try that as it takes care with schema creation at solr.

**Comment by TheOneValen** [comment]
I get 

``` {"success":true,"message":"Full-text index will be re-created."} ```

But search is not working :(

**Comment by eikek** [comment]
Can you post the logs of joex when this is submitted?

**Comment by TheOneValen** [comment]
This says joex:

```

[ioapp-compute-1] DEBUG d.j.s.SchedulerImpl - Forking job BZX92vEqD.../docspell-system/full-text-reindex/Low
[ioapp-compute-1] DEBUG d.j.s.SchedulerImpl - Try to acquire permit (0 free)
[ioapp-compute-1] DEBUG d.j.s.SchedulerImpl - Starting task now
[ioapp-compute-0] INFO  d.j.s.LogSink - >>> 2021-03-22T21:02:45.784842Z Info BZX92vEqD.../docspell-system/full-text-reindex/Low: Running full-text re-index now                                                                                                                            
[ioapp-compute-3] INFO  d.j.s.LogSink - >>> 2021-03-22T21:02:45.785344Z Info BZX92vEqD.../docspell-system/full-text-reindex/Low: Clearing index data                                                                                                                                       
[ioapp-compute-3] INFO  d.j.s.LogSink - >>> 2021-03-22T21:02:45.785958Z Info BZX92vEqD.../docspell-system/full-text-reindex/Low: Deleting complete full-text index!                                                                                                                        
[ioapp-compute-0] INFO  d.j.s.LogSink - >>> 2021-03-22T21:02:45.864966Z Info BZX92vEqD.../docspell-system/full-text-reindex/Low: Running index initialize                                                                                                                                  
[ioapp-compute-3] ERROR d.j.s.SchedulerImpl - Job BZX92vEqD.../docspell-system/full-text-reindex/Low execution failed. Retrying later.
org.http4s.client.UnexpectedStatus: unexpected HTTP status: 400 Bad Request
[ioapp-compute-3] DEBUG d.j.s.SchedulerImpl - Job BZX92vEqD.../docspell-system/full-text-reindex/Low done Stuck. Releasing resources.
[ioapp-compute-0] ERROR d.j.s.LogSink - >>> 2021-03-22T21:02:46.486965Z Error BZX92vEqD.../docspell-system/full-text-reindex/Low: Job BZX92vEqD.../docspell-system/full-text-reindex/Low execution failed. Retrying later.                                                                 
org.http4s.client.UnexpectedStatus: unexpected HTTP status: 400 Bad Request
[ioapp-compute-3] DEBUG d.j.s.SchedulerImpl - Permit released (0 free)
[ioapp-compute-1] DEBUG d.j.s.SchedulerImpl - New permit acquired
[ioapp-compute-1] DEBUG d.j.s.SchedulerImpl - Next job found: None
[ioapp-compute-1] DEBUG d.j.s.SchedulerImpl - Waiting for notify

```

**Comment by eikek** [comment]
Seems that solr is in some unhappy state. Can you also post the logs from solr? Thanks!

I think it might be necessary to re-create the core and run this job again with a new core.

**Comment by TheOneValen** [comment]
```
2021-03-23 19:36:53.059 INFO  (qtp428696898-22) [   x:docspell] o.a.s.u.p.LogUpdateProcessorFactory [docspell]  webapp=/solr path=/update params={commit=true&overwrite=true&wt=json}{deleteByQuery=*:* (-1695052687525019648),commit=} 0 12
2021-03-23 19:36:53.119 INFO  (qtp428696898-23) [   x:docspell] o.a.s.s.ManagedIndexSchema Upgraded to managed schema at /var/solr/data/docspell/conf/managed-schema                                                                                                                       
2021-03-23 19:36:53.137 INFO  (qtp428696898-23) [   x:docspell] o.a.s.c.SolrConfig Using Lucene MatchVersion: 8.8.1                                                                                                                                                                         
2021-03-23 19:36:53.259 INFO  (qtp428696898-23) [   x:docspell] o.a.s.s.IndexSchema Schema name=default-config                                                                                                                                                                             
2021-03-23 19:36:53.266 WARN  (qtp428696898-23) [   x:docspell] o.a.s.s.AbstractSpatialPrefixTreeFieldType FieldType SpatialRecursivePrefixTreeFieldType does not allow omitNorms to be specified in schema, hardcoded behavior is omitNorms=true                                          
2021-03-23 19:36:53.267 WARN  (qtp428696898-23) [   x:docspell] o.a.s.s.AbstractSpatialPrefixTreeFieldType FieldType SpatialRecursivePrefixTreeFieldType does not allow omitTermFreqAndPositions to be specified in schema, hardcoded behavior is omitTermFreqAndPositions=true            
2021-03-23 19:36:53.267 WARN  (qtp428696898-23) [   x:docspell] o.a.s.s.AbstractSpatialPrefixTreeFieldType FieldType SpatialRecursivePrefixTreeFieldType does not allow termOffsets to be specified in schema, hardcoded behavior is termOffsets=false                                     
2021-03-23 19:36:53.267 WARN  (qtp428696898-23) [   x:docspell] o.a.s.s.AbstractSpatialPrefixTreeFieldType FieldType SpatialRecursivePrefixTreeFieldType does not allow termPositions to be specified in schema, hardcoded behavior is termPositions=false                                 
2021-03-23 19:36:53.267 WARN  (qtp428696898-23) [   x:docspell] o.a.s.s.AbstractSpatialPrefixTreeFieldType FieldType SpatialRecursivePrefixTreeFieldType does not allow omitPositions to be specified in schema, hardcoded behavior is omitPositions=true
2021-03-23 19:36:53.269 WARN  (qtp428696898-23) [   x:docspell] o.a.s.s.SchemaManager Unable to persist managed schema.                                                                                                                                                                    
2021-03-23 19:36:53.270 ERROR (qtp428696898-23) [   x:docspell] o.a.s.h.RequestHandlerBase org.apache.solr.api.ApiBag$ExceptionWithErrObject: error processing commands, errors: [Unable to persist managed schema. Unable to reload core [docspell]],
        at org.apache.solr.handler.SchemaHandler.handleRequestBody(SchemaHandler.java:97)                                                                                                                                                                                                   
        at org.apache.solr.handler.RequestHandlerBase.handleRequest(RequestHandlerBase.java:216)                                                
        at org.apache.solr.core.SolrCore.execute(SolrCore.java:2646)                                                                                    
        at org.apache.solr.servlet.HttpSolrCall.execute(HttpSolrCall.java:794)                                                                                             
        at org.apache.solr.servlet.HttpSolrCall.call(HttpSolrCall.java:567)                                                                                                                                                                                                                 
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:427)                                                        
        at org.apache.solr.servlet.SolrDispatchFilter.doFilter(SolrDispatchFilter.java:357)                                                      
```

and many more exceptions.

How can i re-create the core? My deployment uses docker (compose).

**Comment by eikek** [comment]
It looks like some auto-schema kicked in and now prohibits to remove a field.

I think the easiest option is to drop the volume and let it re-create. I'm not a docker/compose expert at all… but I tried this on my machine:

```
docker-compose down
docker container ls -a |grep docspell-solr # copy the id in first column
docker container rm [id]
docker volume rm -f docker_docspell-solr_data
docker-compose up -d

# recreate index
# curl -H 'Docspell-Admin-Secret:123' -XPOST "http://localhost:7880/api/v1/admin/fts/reIndexAll"
```

You need to stop everything, then find out the solr container and remove it, then you can remove the volume. Then start everything up. The solr container is recreated as well as the volume. Then (without trying to search anything), run the re-index task using the admin endpoint. I'm sure there is a quicker way, but as I said, I'm not very experienced with docker things.

**Comment by TheOneValen** [comment]
That did the trick, search works! Thank you. I mapped a path not using a volume, but the procedure is the same (just removing all files from the mounted path). You can just stop and than (re)start the container, too.

There are still some exceptions after calling the admin endpoint: I guess the re index does try to delete some Fields? Those are not there yet, of course.

**Comment by eikek** [comment]
Nice to hear; you're welcome!

> There are still some exceptions after calling the admin endpoint: I guess the re index does try to delete some Fields? Those are not there yet, of course.

Exactly, these exceptions can be ignored.

**Comment by eikek** [comment]
Closing this now. Feel free to re-open or use a new issue any time.

**Comment by fabiosoft** [comment]
same error for me... i use docker-compose.
Even if removed everything and redeployed nothing solved... i needed to give `chmod 777...` on solr folder.

Bad practise but it finally worked.

**Comment by eikek** [comment]
Hi @fabiosoft this is indeed really strange. Did you have to change the folder on your volume? It is necessary that the user running solr has write access to the directory. I never had problems with the solr docker image, but I only use it when playing around.
